src/Database/Relations/HasOne.php

```diff
@@ -43,7 +44,7 @@ public function getResults()
      */
     public function setSimpleValue($value)
     {
-        if (is_array($value)) {
+        if (is_array($value) || $value instanceof Collection) {
```

Comment:
@TimFoerster Instead of this, could you maybe check if $value is iterable?
```suggestion
        if (is_iterable($value)) {
```